### PR TITLE
Update API reference documentation

### PR DESCRIPTION
**Pull request by bravegrape**
Content improvements to ensure consistent language with the technical documentation.

Added logo for minimal branding of the spec.

Added links into the tech docs where applicable.

Interlinked the two specs to make sure users can find their way if they're looking at the wrong spec.

Re-built the static HTML with the updated spec.

### CODE REVIEW

**Comment by bravegrape** [comment]
[Update API reference documentation](https://trello.com/c/if7YZzxN/313-update-api-reference-documentation)

File: docs/api/verify-service-provider-api.swagger.yml [hunk 1]
```diff
@@ -134,14 +144,12 @@ components:
...
-          description: A SAML Response as a base64 encoded string.
+          description: A SAML response as a base64 encoded string.
           type: string
           format: byte
         requestId:
           description: >-
-            A token that was generated for the original SAML Authn Request. This
-            is used to verify that the Authn Request and SAML Response have
-            passed  through the same browser.
+            A token that was generated for the original SAML Authn Request. This is used to verify that the Authn Request and SAML response have passed  through the same browser.
```
Comment (chrisholmes) [inline docs/api/verify-service-provider-api.swagger.yml]: should this be authentication request?

Comment (chrisholmes) [reply docs/api/verify-service-provider-api.swagger.yml]:
```suggestion
            A token that was generated for the original SAML authentication request. This is used to verify that the SAML authentication request and SAML response have passed  through the same browser.
```

File: docs/api/verify-service-provider-api.swagger.yml [hunk 2]
```diff
@@ -167,9 +175,9 @@ components:
           type: string
           enum:
             - AUTHENTICATION_FAILED
-          description: Returned when there was a failure to authenticate.
+          description: Returned when the `urn:oasis:names:tc:SAML:2.0:status:AuthnFailed` status code is included in the SAML response from the GOV.UK Verify Hub.
     NoAuthenticationScenario:
-      description: Returned when there wasn't an authentication
+      description: Returned when the user chose to cancel at some point in their journey.
```
Comment (chrisholmes) [inline docs/api/verify-service-provider-api.swagger.yml]: We should specify the SAML status code here.

Also, this won't probably won't happen due to a cancellation. It's more likely to happen due to a timeout.

File: docs/api/verify-service-provider-with-matching-api.swagger.yml [hunk 3]
```diff
@@ -196,14 +199,14 @@ components:
...
-      description: Returned when the user has cancelled 
+      description: Returned when the user has cancelled
       type: object
       properties:
         scenario:
           type: string
           enum:
             - CANCELLATION
-          description: Returned when the user has cancelled 
+          description: Returned when the user has cancelled
```
Comment (chrisholmes) [inline docs/api/verify-service-provider-with-matching-api.swagger.yml]: We can use the SAML status code here too.

File: docs/api/verify-service-provider-api.swagger.yml (hunk 2 above)
Comment (chrisholmes) [reply docs/api/verify-service-provider-api.swagger.yml]: It's `urn:oasis:names:tc:SAML:2.0:status:NoAuthnContext` FYI

File: docs/api/verify-service-provider-with-matching-api.swagger.yml [hunk 4]
```diff
@@ -223,20 +226,17 @@ components:
             - AUTHENTICATION_FAILED
           description: Returned when there was a failure to authenticate.
```
Comment (chrisholmes) [inline docs/api/verify-service-provider-with-matching-api.swagger.yml]: SAML status code here?

File: docs/api/verify-service-provider-api.swagger.yml (hunk 2 above)
Comment (bravegrape) [reply docs/api/verify-service-provider-api.swagger.yml]:
```suggestion
      description: Returned when the user chose to cancel at some point in their journey. The SAML response for the GOV.UK Verify Hub contains the `urn:oasis:names:tc:SAML:2.0:status:NoAuthnContext` status code.
```

File: docs/api/verify-service-provider-with-matching-api.swagger.yml (hunk 4 above)
Comment (bravegrape) [reply docs/api/verify-service-provider-with-matching-api.swagger.yml]: @chrisholmes could you just check this is the correct status code 🤔 

```suggestion
          description: Returned when there was a failure to authenticate. The SAML response from GOV.UK Verify Hub contains the `urn:oasis:names:tc:SAML:2.0:status:AuthnFailed` status code.
```

File: docs/api/verify-service-provider-with-matching-api.swagger.yml (hunk 3 above)
Comment (bravegrape) [reply docs/api/verify-service-provider-with-matching-api.swagger.yml]: @chrisholmes can you check I got the right status code here:

```suggestion
          description: Returned when the user has cancelled.  The SAML response for the GOV.UK Verify Hub contains the `urn:oasis:names:tc:SAML:2.0:status:Requester` status code.

```

File: docs/api/verify-service-provider-api.swagger.yml [hunk 5]
```diff
@@ -167,9 +175,9 @@ components:
           type: string
           enum:
             - AUTHENTICATION_FAILED
-          description: Returned when there was a failure to authenticate.
+          description: Returned when the `urn:oasis:names:tc:SAML:2.0:status:AuthnFailed` status code is included in the SAML response from the GOV.UK Verify Hub.
     NoAuthenticationScenario:
-      description: Returned when there wasn't an authentication
+      description: Returned when the user chose to cancel at some point in their journey. The SAML response for the GOV.UK Verify Hub contains the `urn:oasis:names:tc:SAML:2.0:status:NoAuthnContext` status code.
```
Comment (chrisholmes) [inline docs/api/verify-service-provider-api.swagger.yml]:
```suggestion
      description: Returned when the `urn:oasis:names:tc:SAML:2.0:status:NoAuthnContext` status code is included in the SAML response from the GOV.UK Verify Hub.
```

File: docs/api/verify-service-provider-api.swagger.yml [hunk 6]
```diff
@@ -178,7 +186,7 @@ components:
           type: string
           enum:
             - NO_AUTHENTICATION
-          description: Returned when there wasn't an authentication
+          description: Returned when the user chose to cancel at some point in their journey.
```
Comment (chrisholmes) [inline docs/api/verify-service-provider-api.swagger.yml]:
```suggestion
          description: Returned when the `urn:oasis:names:tc:SAML:2.0:status:NoAuthnContext` status code is included in the SAML response from the GOV.UK Verify Hub.
```

**Pull request merged**